crates/storage_api2/src/lib.rs

```diff
@@ -78,12 +81,14 @@ pub trait StorageApi: Send + Sync + 'static {
     fn execute(
         &self,
         operation: Operation<'_>,
-    ) -> impl Future<Output = Result<operation::Output>> + Send;
+    ) -> impl Future<Output = Result<operation::Output>> + Send {
+        async move { self.execute_ref(&operation).await }
```

Comment:
So the default implementation of the trait would result in a stack overflow due to recursion? Not sure it's a good idea.

Reply:
This is a neat trick I didn't know about either. Stack overflow is not possible when dealing with statically typed futures. Compiler is smart enough to throw an error forcing you to implement one of the functions.